### PR TITLE
CLOUDP-167772: performance advisor commands

### PR DESCRIPTION
**Pull request by wtrocki**
Migration commands for performance advisor

### CODE REVIEW

**Comment by wtrocki** [comment]
[CLOUDP-167771-events](https://spruce.mongodb.com/task/mongodb_atlas_cli_master_e2e_atlas_clusters_atlas_performance_advisor_e2e_patch_7e4b07475241cc10c2b2cb4aef7917e6fe194179_6447b42f61837d414c649323_23_04_25_11_06_24/logs?execution=0)

File: internal/cli/atlas/performanceadvisor/namespaces/list.go [hunk 1]
```diff
@@ -54,18 +54,20 @@ func (opts *ListOpts) Run() error {
...
-func (opts *ListOpts) newNamespaceOptions() *atlas.NamespaceOptions {
-	return &atlas.NamespaceOptions{
-		Since:    opts.since,
-		Duration: opts.duration,
+func (opts *ListOpts) newNamespaceOptions(project, host string) *atlas.ListSlowQueryNamespacesApiParams {
+	return &atlas.ListSlowQueryNamespacesApiParams{
+		GroupId:   project,
+		ProcessId: host,
+		Since:     &opts.since,
+		Duration:  &opts.duration,
```
Comment (matt-condon) [inline internal/cli/atlas/performanceadvisor/namespaces/list.go:70]: 😮

Comment (matt-condon) [inline internal/cli/atlas/performanceadvisor/namespaces/list.go:70]: 😮

File: internal/cli/atlas/performanceadvisor/suggestedindexes/list.go [hunk 2]
```diff
@@ -57,23 +57,26 @@ func (opts *ListOpts) Run() error {
...
-	return &atlas.SuggestedIndexOptions{
-		Namespaces: opts.namespaces,
-		NIndexes:   opts.nIndexes,
-		NExamples:  opts.nExamples,
-		NamespaceOptions: atlas.NamespaceOptions{
-			Since:    opts.since,
-			Duration: opts.duration,
-		},
+func (opts *ListOpts) newSuggestedIndexOptions(project, host string) *atlas.ListSuggestedIndexesApiParams {
+	// TODO log issue with the API
```
Comment (matt-condon) [inline internal/cli/atlas/performanceadvisor/suggestedindexes/list.go:69]: Still required?

**Review by matt-condon — APPROVED** [review]
LGTM

File: internal/cli/atlas/performanceadvisor/namespaces/list.go (hunk 1 above)
Comment (wtrocki) [reply internal/cli/atlas/performanceadvisor/namespaces/list.go:70]: Not sure what you mean

**Comment by wtrocki** [comment]
All PRs I have have failing e2e tests. Checking that now.

**Comment by gssbzn** [comment]
Detail: Invalid query parameter Duration specified. Reason: Bad Request. Params: [Duration]

File: internal/cli/atlas/performanceadvisor/slowquerylogs/list.go [hunk 3]
```diff
@@ -25,7 +25,7 @@ import (
 	store "github.com/mongodb/mongodb-atlas-cli/internal/store/atlas"
 	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
 	"github.com/spf13/cobra"
-	atlas "go.mongodb.org/atlas/mongodbatlas"
+	atlas "go.mongodb.org/atlas/mongodbatlasv2"
```
Comment (swattyT) [inline internal/cli/atlas/performanceadvisor/slowquerylogs/list.go:28]: atlasv2

File: internal/cli/atlas/performanceadvisor/namespaces/list.go (hunk 1 above)
Comment (matt-condon) [reply internal/cli/atlas/performanceadvisor/namespaces/list.go:70]: I assume based on the new params, this is the new change to the v2 SDK to pass the params as a struct. 
Nice improvement!

File: internal/cli/atlas/performanceadvisor/suggestedindexes/list.go (hunk 2 above)
Comment (wtrocki) [reply internal/cli/atlas/performanceadvisor/suggestedindexes/list.go:69]: Good catch. Done already (forgot to include jira).

**Comment by wtrocki** [comment]
This should be now unblocked

**Comment by wtrocki** [comment]
Integrations tests are passing

**Review by swattyT — APPROVED** [review]
Added atlas_performance_advisor_e2e  and atlas_gov_performance_advisor_e2e  to the tests.
LGTM if the tests are good. Thanks!

**Pull request merged**